Execute Permissions Not Set After Java Runtime Download on Linux

When running on Linux systems, the bootstrap downloads the Java runtime successfully but fails to set the appropriate execute permissions. This causes the launcher to fail when trying to start, requiring manual intervention (`chmod +x`) from the user.

#### Current Behavior
1. Bootstrap downloads Java runtime
2. Runtime files are extracted
3. Launcher fails to start with permission denied error
4. User needs to manually run `chmod +x` on the Java executable

#### Expected Behavior
The bootstrap should automatically set the correct execute permissions after downloading and extracting the Java runtime, allowing the launcher to start without manual intervention.

#### Error Message
```
java.io.IOException: Cannot run program "/.altiscore/runtime/bin/java" (in directory "/.altiscore"): error=13, Permission denied
    at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1143)
    at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1073)
    at fr.arinonia.bootstrap.launcher.LauncherStarter.startLauncher(LauncherStarter.java:43)
    [...]
Caused by: java.io.IOException: error=13, Permission denied
    at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
```

#### Steps to Reproduce
1. Run bootstrap on Linux system
2. Let it download and extract Java runtime
3. Wait for launcher to start
4. Observe permission denied error

#### Environment
- OS: Linux
- Java Version: 21
- Bootstrap Version: 1.0.0

#### Proposed Solution
Add permission setting in the `RuntimeDownloader` or `FileManager` class after extraction:
```java
Path javaExecutable = downloadPath.resolve("bin").resolve("java");
Files.setPosixFilePermissions(javaExecutable, 
    PosixFilePermissions.fromString("rwxr-xr-x"));
```

#### Additional Context
This issue only affects Linux systems as Windows handle executable permissions differently. The current workaround is to manually run:
```bash
chmod +x ~/.altiscore/runtime/bin/java
```

#### Checklist
- [ ] Set execute permissions on Java executable
- [ ] Test on different Linux distributions
- [ ] Consider adding permission verification before launcher start

Label: `bug`

--- a/build.gradle
+++ b/build.gradle
@@ -5,7 +5,7 @@ plugins {
 }
 
 group 'fr.arinonia'
-version '1.0-SNAPSHOT'
+version '1.0.1'
 
 repositories {
     mavenCentral()

--- a/src/main/java/fr/arinonia/bootstrap/config/BootstrapConfig.java
+++ b/src/main/java/fr/arinonia/bootstrap/config/BootstrapConfig.java
@@ -11,7 +11,7 @@ public class BootstrapConfig {
 
     // Application
     private final String appName = "AltisCore Bootstrap";
-    private final String appVersion = "1.0.0";
+    private final String appVersion = "1.0.1";
     private final String discordUrl = "https://discord.gg/your-discord";
 
     // Runtime

--- a/src/main/java/fr/arinonia/bootstrap/file/FileManager.java
+++ b/src/main/java/fr/arinonia/bootstrap/file/FileManager.java
@@ -2,10 +2,14 @@ package fr.arinonia.bootstrap.file;
 
 import fr.arinonia.bootstrap.config.BootstrapConfig;
 import fr.arinonia.bootstrap.utils.OSDetector;
+import fr.flowarg.azuljavadownloader.AzulJavaOS;
 
 import java.io.IOException;
 import java.nio.file.*;
+import java.nio.file.attribute.PosixFilePermission;
 import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Set;
 import java.util.stream.Stream;
 
 public class FileManager {
@@ -69,6 +73,8 @@ public class FileManager {
                 });
             }
 
+            setExecutablePermissions();
+
             try (final Stream<Path> pathsToDelete = Files.walk(extractedDir)
                     .sorted(Comparator.reverseOrder())) {
                 for (final Path path : (Iterable<Path>) pathsToDelete::iterator) {
@@ -82,6 +88,42 @@ public class FileManager {
                 }
             }
         }
+    }
+
+    private void setExecutablePermissions() {
+        if (!OSDetector.isUnixSystem()) {
+            System.out.println("Skipping setting executable permissions for non-Linux platforms");
+            return;
+        }
+        System.out.println("Setting executable permissions for runtime binaries...");
+        final String[] executableFiles = {
+                "bin/java",
+                "bin/keytool",
+                "bin/jpackage",
+                "lib/jspawnhelper"
+        };
+
+        final Set<PosixFilePermission> permissions = new HashSet<>();
+        permissions.add(PosixFilePermission.OWNER_READ);
+        permissions.add(PosixFilePermission.OWNER_WRITE);
+        permissions.add(PosixFilePermission.OWNER_EXECUTE);
+        permissions.add(PosixFilePermission.GROUP_READ);
+        permissions.add(PosixFilePermission.GROUP_EXECUTE);
+        permissions.add(PosixFilePermission.OTHERS_READ);
+        permissions.add(PosixFilePermission.OTHERS_EXECUTE);
+
+        for (final String execFile : executableFiles) {
+            final Path filePath = this.runtimePath.resolve(execFile);
+            if (Files.exists(filePath)) {
+                try {
+                    Files.setPosixFilePermissions(filePath, permissions);
+                    System.out.println("Set executable permissions for: " + filePath);
+                } catch (final IOException e) {
+                    System.err.println("Warning: Could not set permissions for " + filePath + ": " + e.getMessage());
+                }
+            }
+        }
+
     }
 
     public Path getRootPath() {

--- a/src/main/java/fr/arinonia/bootstrap/utils/OSDetector.java
+++ b/src/main/java/fr/arinonia/bootstrap/utils/OSDetector.java
@@ -62,4 +62,13 @@ public class OSDetector {
             return false;
         }
     }
+
+    /**
+     * Check if the current system is a Unix system
+     * @return true if the current system is a Unix system
+     */
+    public static boolean isUnixSystem() {
+        final String os = System.getProperty("os.name").toLowerCase();
+        return os.contains("nix") || os.contains("nux") || os.contains("aix");
+    }
 }

